src/services/audio/audio.service.ts

```diff
@@ -17,10 +17,12 @@ export class AudioService {
         return Observable.create((observer: Observer<AudioMetadata>) => {
             this._http.get(mediaElement.src, { responseType: 'blob' }).subscribe(response => {
 
-                const filename = mediaElement.src.substring(mediaElement.src.lastIndexOf('/') + 1);
                 const extension = mediaElement.src.substring(mediaElement.src.lastIndexOf('.') + 1).toLowerCase();
-
+                let filename;
                 let description;
+                let sourceString: string = mediaElement.src;
+
+                sourceString.includes('base64') ? filename = '' : filename = sourceString.substring(mediaElement.src.lastIndexOf('/') + 1);
```

Comment:
This can be simplified, we shouldn't need to duplicate the `filename = ` part and can get rid of `sourceString`.
We should also avoid using `includes` as this in not supported in all browsers and is one of the functions that is not always polyfilled by default:

```typescript
const filename = mediaElement.src.indexOf('base64') !== -1 ? '' : mediaElement.src.substring(mediaElement.src.lastIndexOf('/') + 1)
```